Tribler/community/gigachannel/community.py

```diff
@@ -92,13 +111,67 @@ def on_blob(self, peer, blob):
         if reply_blob:
             self.endpoint.send(peer.address, self.ezr_pack(self.NEWS_PUSH_MESSAGE, RawBlobPayload(reply_blob)))
 
+    # def create_search_request(self, query_filter, metadata_type='', sort_by="HEALTH", sort_asc=0, hide_xxx=True):
+    #     cache = self.request_cache.add(SearchRequest(self.request_cache, query_filter))
+    #     self._logger.info("Sending search query:%s, identifier:%s", query_filter, cache.number)
+    #
+    #     return SearchRequestPayload(cache.number, query_filter, metadata_type, sort_by, sort_asc, hide_xxx)
+
+    def send_search_request(self, query_filter, metadata_type='', sort_by="HEALTH", sort_asc=0, hide_xxx=True):
+        self.search_counter += 1
+        request = SearchRequestPayload(self.search_counter, query_filter, metadata_type, sort_by, sort_asc, hide_xxx)
+        self._logger.debug("Started remote search for query:%s", query_filter)
+
+        for peer in self.get_peers()[:max_search_peers]:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_REQUEST, request))
+
+    @lazy_wrapper(SearchRequestPayload)
+    def on_search_request(self, peer, request):
+        unicode_query_filter = cast_to_unicode_utf8(request.query_filter)
+        metadata_type = {'': [REGULAR_TORRENT, CHANNEL_TORRENT],
+                         "channel": CHANNEL_TORRENT,
+                         "torrent": REGULAR_TORRENT
+                         }.get(request.metadata_type, REGULAR_TORRENT)
+
+        request_dict = {
+            "first": 1,
+            "last": max_entries,
+            "sort_by": request.sort_by,
+            "sort_asc": request.sort_asc,
+            "query_filter": unicode_query_filter,
+            "hide_xxx": request.hide_xxx,
+            "metadata_type": metadata_type
+        }
+
+        with db_session:
+            db_results, total = self.metadata_store.TorrentMetadata.get_entries(**request_dict)
+            if total > 0:
+                blob = entries_to_chunk(db_results[:max_entries], maximum_payload_size)[0]
+
+        self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_RESPONSE, SearchResponsePayload(request.id, blob)))
+
+    @lazy_wrapper(SearchResponsePayload)
+    def on_search_response(self, peer, response):
+        if self.search_counter != response.id:
+            return
+        with db_session:
+            try:
+                metadata_result = self.metadata_store.process_compressed_mdblob(response.raw_blob)
+            except RuntimeError as re:
+                self.logger.error("Could not process mdblob: %s", re)
+                return
+
+            search_results = [(dict(type={REGULAR_TORRENT: 'torrent', CHANNEL_TORRENT: 'channel'}[r.metadata_type],
+                                    **(r.to_simple_dict()))) for (r, n) in metadata_result]
+        self.notifier.notify(SIGNAL_GIGACHANNEL_COMMUNITY, SIGNAL_ON_SEARCH_RESULTS, search_results, '')
```

Comment:
`self.notifier` can be `None`, please check for that.

Reply:
Good point, done